File: python/pe007.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 007 """

import sys
from common import is_prime


PRIMENUM = 10001
primes = [2, 3]

test = primes[-1] + 2
while len(primes) < PRIMENUM:
    if is_prime(test, primes):
        primes.append(test)
    test += 2

print(primes[-1])
sys.exit(0)

File: lua/pe007.lua
-- this computes projecteuler.net problem 007

local function is_prime(test, primes) 
  root = math.sqrt(test)

  for i, prime in ipairs(primes) do
    if prime > root then
      return true
    end
    if (test % prime) == 0 then
      return false
    end
  end
end

local primenum = 10001
local primes = {2,3}

local test = 5
local counter = 3
while counter <= primenum do
  if is_prime(test, primes) then
    primes[counter] = test
    counter = 1 + counter
  end
  test = 2 + test
end

print(primes[counter - 1])
